mitodl/mitxpro
# feat: display tax in regular checkout

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes https://github.com/mitodl/mitxpro/issues/2769

#### What's this PR do?

- Apply taxes on the checkout page.
- Display taxes in Email Receipt and Dashboard Receipt View.

#### How should this be manually tested?

_NOTE: I will write tests for the frontend in a followup PR. Also, I was not able to test the Cybersource checkout as my Payments were declined each time. I tested with the discounts then I tested the Receipt by deleting Coupon redemptions._

- Create a couple of courses and products.
- Try to enroll and Verify that the tax is calculated and it is as per the tax rate. Tax should be visible in the checkout order summary as per the [designs](https://www.figma.com/proto/140ipJMKt1up7wHA0OUoXY/xPRO---VAT-Address?page-id=0%3A1&node-id=43-11&scaling=min-zoom&mode=design&t=iHbE356DwLYM1moB-1).
- Proceed with the payment in Cybersource.
- Once enrollment is complete, Verify that the Email Receipt has `Tax` listed as per the [designs](https://www.figma.com/proto/140ipJMKt1up7wHA0OUoXY/xPRO---VAT-Address?page-id=0%3A1&node-id=110-33&scaling=min-zoom&mode=design&t=iHbE356DwLYM1moB-1) and the values are correct.
- View Receipt in the Dashboard and verify the Tax and other details as per the [designs](https://www.figma.com/proto/140ipJMKt1up7wHA0OUoXY/xPRO---VAT-Address?page-id=0%3A1&node-id=7-8&scaling=min-zoom&mode=design&t=iHbE356DwLYM1moB-1).
- Now repeat the same process after applying some discount codes.

#### Screenshots (if appropriate)

Checkout:
<img width="1574" alt="Screen Shot 2023-09-29 at 7 45 04 PM" src="https://github.com/mitodl/mitxpro/assets/52656433/386a96aa-88cb-4e3b-886a-f3d90756b504">


Email:
<img width="1342" alt="Screen Shot 2023-09-29 at 7 46 13 PM" src="https://github.com/mitodl/mitxpro/assets/52656433/6c94d4bf-362c-4e4a-baef-94182a16dfca">

Dashboard Receipt:
<img width="1791" alt="Screen Shot 2023-09-29 at 7 45 41 PM" src="https://github.com/mitodl/mitxpro/assets/52656433/520fe9ac-1328-4971-aaed-bf26883a744f">
<img width="1791" alt="Screen Shot 2023-09-29 at 7 45 52 PM" src="https://github.com/mitodl/mitxpro/assets/52656433/cde49afc-b017-48ce-8991-86cb9769a3e5">
